AnkiDroid/src/test/java/com/ichi2/anki/analytics/AnalyticsConstantsTest.java

```diff
@@ -0,0 +1,141 @@
+            listOfConstantFields.add("Exception Report");
+            listOfConstantFields.add("aedict");
+            listOfConstantFields.add("leo");
+            listOfConstantFields.add("colordict");
+            listOfConstantFields.add("fora");
+            listOfConstantFields.add("nciku");
+            listOfConstantFields.add("eijiro");
+            return listOfConstantFields;
+        }
+
+
+        /**
+         * The message here means that the string being checked cannot be found in Actions class.
+         * If encountered with this message, re-check the list present here and constants in Actions class, to resolve
+         * the test failure.
+         */
+        @Test
+        public void checkAnalyticsString() {
+            assertEquals("There is no such string in Actions class, thus returning null.", analyticsString, getStringFromReflection(analyticsString));
+        }
+
+
+        public String getStringFromReflection(String analyticsStringToBeChecked) {
+            String reflectedString = null;
+            UsageAnalytics.Actions actions = new UsageAnalytics.Actions();
+            Field[] field;
+            field = actions.getClass().getDeclaredFields();
+            for (Field value : field) {
+                if (value.isAnnotationPresent(AnalyticsConstant.class)) {
+                    try {
+                        if (value.get(actions).equals(analyticsStringToBeChecked)) {
+                            reflectedString = (String) value.get(actions);
+                        }
+                    } catch (IllegalAccessException e) {
+                        throw new RuntimeException();
+                    }
+                }
+            }
+            return reflectedString;
+        }
+
+    }
+
+
+    public static class AnalyticsConstantsFieldLengthTest {
+
+        @Test
+        public void fieldSizeEqualsListOfConstantFields() {
+            if (getFieldSize() > listOfConstantFields.size()) {
+                assertEquals("Add the new constant here also in the list listOfConstantFields", listOfConstantFields.size(), getFieldSize());
+            } else if (getFieldSize() < listOfConstantFields.size()) {
+                assertEquals("Remove the constant from the list listOfConstantFields", listOfConstantFields.size(), getFieldSize());
+            } else {
+                assertEquals(listOfConstantFields.size(), getFieldSize());
+            }
+
+        }
+
+
+        /**
+         * This method is used to get the size of fields in Actions Class.
+         * Because whenever a new constant is added in Actions Class but not added to the list present in this
+         * class (listOfConstantFields) the test must fail.
+         */
+        public static int getFieldSize() {
+            UsageAnalytics.Actions actions = new UsageAnalytics.Actions();
+            Field[] field;
+            field = actions.getClass().getDeclaredFields();
+            return field.length;
+        }
```

Comment:
This is untested (so therefore lazy). I need to get CI working on my repo. If this doesn't work, I'll put in more effort.

```suggestion
        public static long getFieldSize() {
            return Arrays.stream(UsageAnalytics.class.getDeclaredFields())
                    .filter(x -> x.isAnnotationPresent(AnalyticsConstant.class))
                    .count();
        }
```

Reply:
> One thing is missing here. ( `UsageAnalytics.Actions.class` )

🤦 yes, that's what I get for laziness, thanks!